sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/internal/DefaultConfigProperties.java

```diff
@@ -27,26 +27,33 @@
 import javax.annotation.Nullable;
 
 /**
- * Properties to be used for auto-configuration of the OpenTelemetry SDK components. These
- * properties will be a combination of system properties and environment variables. The properties
- * for both of these will be normalized to be all lower case, and underscores will be replaced with
+ * Properties are normalized to The properties for both of these will be normalized to be all lower
+ * case, dashses are replaces with periods, and environment variable underscores are replaces with
  * periods.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
  */
-final class DefaultConfigProperties implements ConfigProperties {
+public final class DefaultConfigProperties implements ConfigProperties {
 
   private final Map<String, String> config;
 
-  static DefaultConfigProperties get(Map<String, String> defaultProperties) {
+  /**
+   * Creates a {@link DefaultConfigProperties} by merging system properties, environment variables,
+   * and the {@code defaultProperties}.
+   *
+   * <p>Environment variables take priority over {@code defaultProperties}. System properties take
+   * priority over system properties.
+   */
+  public static DefaultConfigProperties create(Map<String, String> defaultProperties) {
     return new DefaultConfigProperties(System.getProperties(), System.getenv(), defaultProperties);
   }
 
-  static DefaultConfigProperties customize(
-      DefaultConfigProperties previousProperties, Map<String, String> overrides) {
-    return new DefaultConfigProperties(previousProperties, overrides);
-  }
```

Comment:
Adjusted this to be an instance method called `withOverrides`. Also renamed `static DefaultConfigProperties get` to `create`, so we now have consistently named factory methods for creating instances.